src/client/Microsoft.Identity.Client/Internal/JsonWebToken.cs

```diff
@@ -65,18 +70,17 @@ private string CreateJsonPayload()
                 payload.Append('{');
             }
 
-            int i = 0;
-            foreach (var kvp in _claimsToSign)
+            var json = new JObject();
+
+            foreach (var claim in _claimsToSign)
             {
-                payload.Append($"\"{kvp.Key}\":\"{kvp.Value}\"");
```

Comment:
What was the reason for switching this to manual string parsing? Performance right? Since we are no longer parsing the default claims with JSON, but only the custom claims, I would imagine this approach will keep some of the performance gains depending on how many claims are added by the client app.

The reason I went with this approach is the JSON api will handle all of the edge cases we may miss trying to parse manually.

Reply:
SOunds good, thanks Travis. Just add some unit tests for this case, they are clearly missing.